hub/core/storage/provider.py

```diff
@@ -102,3 +152,6 @@ def __len__(self):
         Returns:
             int: the number of files present inside the root
         """
+
+    def flush(self):
+        """flush"""
```

Comment:
What does this do? I thought that `__setitem__` went directly to the store

Reply:
Yeah it does, it's just that LRUCache also inherits this class and I kept this as a no op in case of regular providers. For LRUCache, flush actually does the work. We can remove this as well, but will just need to add a check while flushing anything to see if the storage has flush or not.